src/app/views/forgotten-password/setForgottenPasswordController.jsx

```diff
@@ -11,12 +15,115 @@ export class SetForgottenPasswordController extends Component {
     constructor(props) {
         super(props);
         this.state = {
-            hasSubmitted: false
+            hasSubmitted: false,
```

Comment:
I don't think you need all that on state, you can make constant outside the class so it is not created on every render. 
`const STATUS = { SUBMITTED : 'SUBMITTED',  SUBMITTING: 'SUBMITTING', .... etc } `
   and then when you modify just reference `this.setState({ STATUS.SUBMITTED });` and state would only hold one status at the same time , you also probably need completed etc .. and idle?

Reply:
We have been using `hasSubmitted` and `isSubmitting` in other places, but yes I think you are right it would be better to use an enumerator to store the state as it will never be both `hasSubmitted` and `isSubmitted` at the same time.

Reply:
Like it, this has now been done